tests/bin/check_tron_jobs_test.py

```diff
@@ -1281,3 +1287,153 @@ def test_guess_realert_every_first_time_job(self):
         }
         realert_every = check_tron_jobs.guess_realert_every(job_runs)
         assert_equal(realert_every, -1)
+
+
+class TestCheckPreciousJobs(TestCase):
+    @setup
+    def setup_job(self):
+        self.job_name = 'fake_job'
+        self.monitoring = {
+            'team': 'fake_team',
+            'notification_email': 'fake_email',
+            _PRECIOUS_RUNS_ATTR: True,
+        }
+        self.runs = [
+            {
+                'id': f"{self.job_name}.0",
+                'job_name': self.job_name,
+                'run_num': 0,
+                'run_time': '2018-10-10 12:00:00',
+                'start_time': '2018-10-10 12:00:00',
+                'end_time': '2018-10-10 12:30:00',
+                'state': 'failed',
+                'exit_status': 1,
+            }, {
+                'id': f"{self.job_name}.1",
+                'job_name': self.job_name,
+                'run_num': 1,
+                'run_time': '2018-10-10 13:00:00',
+                'start_time': '2018-10-10 13:00:00',
+                'end_time': '2018-10-10 13:30:00',
+                'state': 'succeeded',
+                'exit_status': 0,
+            }, {
+                'id': f"{self.job_name}.2",
+                'job_name': self.job_name,
+                'run_num': 2,
+                'run_time': '2018-10-11 12:00:00',
+                'start_time': '2018-10-11 12:00:00',
+                'end_time': '2018-10-11 12:30:00',
+                'state': 'succeeded',
+                'exit_status': 0,
+            }, {
+                'id': f"{self.job_name}.3",
+                'job_name': self.job_name,
+                'run_num': 3,
+                'run_time': '2018-10-11 13:00:00',
+                'start_time': '2018-10-11 13:00:00',
+                'end_time': '2018-10-11 13:30:00',
+                'state': 'failed',
+                'exit_status': 1,
+            }, {
+                'id': f"{self.job_name}.4",
+                'job_name': self.job_name,
+                'run_num': 4,
+                'run_time': '2018-10-12 12:00:00',
+                'start_time': '2018-10-12 12:00:00',
+                'end_time': '2018-10-12 12:30:00',
+                'state': 'failed',
```

Comment:
Could you add the tests that the job run becomes unknown, or running longer than expected_runtime? I think the code can handle that, but it's nice to make sure that again.

Reply:
Things to do with state live in `compute_check_result_for_job_runs`, which I have not meaningfully touched. After all, my changes aren't really concerned with state, just how we bucket runs. There are already plenty of tests for `compute_check_result_for_job_runs`, including for unknown states, so I don't think adding further tests for precious runs involving unknown is really necessary.